tests/migrations/test_operations.py

```diff
@@ -1380,52 +1380,39 @@ def test_add_field(self):
         self.assertEqual(sorted(definition[2]), ["field", "model_name", "name"])
 
     @skipUnlessDBFeature("supports_stored_generated_columns")
-    def test_add_generate_field(self):
-        app_label = "test_add_generate_field"
-        project_state = self.apply_operations(
-            app_label,
-            ProjectState(),
-            operations=[
-                migrations.CreateModel(
-                    "Rider",
-                    fields=[
-                        ("id", models.AutoField(primary_key=True)),
-                    ],
+    def test_add_generate_field_after_adding_fk(self):
+        app_label = "test_igfpc"
+        project_state = self.set_up_test_model(app_label)
+        operations = [
+            migrations.CreateModel(
+                "Rider",
+                fields=[
+                    ("id", models.AutoField(primary_key=True)),
+                ],
+            ),
+            migrations.AddField(
+                "Pony",
+                "rider",
+                models.ForeignKey(
+                    "Rider", models.SET_NULL, db_column="rider_id", null=True
                 ),
-                migrations.CreateModel(
-                    "Pony",
-                    fields=[
-                        ("id", models.AutoField(primary_key=True)),
-                        ("name", models.CharField(max_length=20)),
-                        (
-                            "rider",
-                            models.ForeignKey(
-                                f"{app_label}.Rider", on_delete=models.CASCADE
-                            ),
-                        ),
-                        (
-                            "name_and_id",
-                            models.GeneratedField(
-                                expression=Concat(("name"), ("rider_id")),
-                                output_field=models.TextField(),
-                                db_persist=True,
-                            ),
-                        ),
-                    ],
+            ),
+            migrations.AddField(
+                "Pony",
+                "modified_pink",
+                models.GeneratedField(
+                    expression=F("pink") + F("pink"),
```

Comment:
I think this should use `rider_id` to be a regression test for ticket-35359 :thinking: 
```suggestion
                    expression=F("pink") + F("rider_id"),
```

Reply:
Updated 👍

Reply:
So, the original test `OperationTests.test_add_generate_field` is "extra" (does not fail when the changes of [ticket-35359](https://code.djangoproject.com/ticket/35359) is reverted and so isn't a regression test). 
Only confirms that we can add GeneratedFields after adding FKs.

Another option is remove the test entirely.